src/plugins/relay/weechat/relay-weechat-protocol.c

```diff
@@ -840,6 +843,68 @@ RELAY_WEECHAT_PROTOCOL_CALLBACK(completion)
     return WEECHAT_RC_OK;
 }
 
+/*
+ * Callback for signals "hotlist_*".
+ */
+
+int
+relay_weechat_protocol_signal_hotlist_cb (const void *pointer, void *data,
+                                          const char *signal,
+                                          const char *type_data,
+                                          void *signal_data)
+{
+    struct t_relay_client *ptr_client;
+    struct t_hdata *ptr_hdata_buffer;
+    struct t_gui_buffer *ptr_buffer;
+    struct t_gui_hotlist *ptr_hotlist;
+    char cmd_hdata[64], str_signal[128];
+    struct t_relay_weechat_msg *msg;
+
+    /* make C compiler happy */
+    (void) data;
+    (void) type_data;
+
+    ptr_client = (struct t_relay_client *)pointer;
+    if (!ptr_client || !relay_client_valid (ptr_client))
+        return WEECHAT_RC_OK;
+
+    snprintf (str_signal, sizeof (str_signal), "_%s", signal);
+
+    if (strcmp (signal, "hotlist_changed") == 0)
+    {
+        ptr_buffer = (struct t_gui_buffer *)signal_data;
+        if (!ptr_buffer)
+            return WEECHAT_RC_OK;
+
+        ptr_hdata_buffer = weechat_hdata_get ("buffer");
+        if (!ptr_hdata_buffer)
+            return WEECHAT_RC_OK;
+
+        ptr_hotlist = weechat_hdata_pointer (ptr_hdata_buffer, ptr_buffer, "hotlist");
+        if (!ptr_hotlist)
+            return WEECHAT_RC_OK;
+
+        /* send signal only if sync with flag "hotlist" */
+        if (relay_weechat_protocol_is_sync (ptr_client, NULL,
+                                            RELAY_WEECHAT_PROTOCOL_SYNC_HOTLIST))
+        {
+            msg = relay_weechat_msg_new (str_signal);
+            if (msg)
+            {
+                snprintf (cmd_hdata, sizeof (cmd_hdata),
+                          "hotlist:0x%p", ptr_hotlist);
```

Comment:
Even if this is equivalent, please use `0x%lx` here, as done in the other functions.

Reply:
Done. I also fixed the creation_time members and removed the next/prev hot list pointers, as they are always null anyway.